File: C/1250.c
#include <stdio.h>

int main() {

        int x = 0;
        scanf("%d", &x);

        for (int _ = 0; _ < x; _++) {
            int tam = 0;
            scanf("%d", &tam);

            int vet[tam];
            for (int i = 0; i < tam; i++) {
                scanf("%d", &vet[i]);
            }

            char entrada[tam];
            scanf("%s",entrada);

            int tiros = 0;

            for (int i = 0; i < tam; i++) {
                if (vet[i] <= 2 && entrada[i] == 'S'){
                    tiros++;
                } else if (vet[i] > 2 && entrada[i] == 'J'){
                    tiros++;
                }
            }

            printf("%d\n",tiros);
        }
    
    return 0;
}

File: Python/1250.py
while True:
    try:
        for _ in range(int(input())):
            tam = input()
            lista = list(map(int, input().split()))
            pulos = input()

            tiros = 0

            for i, j in zip(lista, pulos):
                if j == 'S' and i <= 2:
                    tiros += 1
                elif j == 'J' and i > 2:
                    tiros += 1
            
            print(tiros)
    except EOFError:
        break